lib/puppet/provider/templates/webadministration/_getapppools.ps1.erb

```diff
@@ -0,0 +1,3 @@
+# TODO: this is JSON, PS v2 doesnt support it
+# PSCustomObject isnt present in v2
+Get-WebConfiguration -Filter '/system.applicationHost/applicationPools/add' | ConvertTo-Json -Depth 10
```

Comment:
I was trying to locate these ps1 templates inside each provider to avoid name collision or a massive folder. What do you think?

I advise doing what I did with iis_site and custom crafting the object to be passed to convert-json, as there is a severe performance penalty as well as a potential for errors when passing this directly to that cmdlet.